test/suite/helper_tiered.py

```diff
@@ -65,25 +96,13 @@ def generate_s3_prefix(test_name = ''):
     # Range upto int32_max, matches that of C++'s std::default_random_engine
     prefix += '_' + str(random.randrange(1, 2147483646)) + '--'
 
-    if test_name:
-        prefix += test_name + '--'
+    # If the calling function has not provided a name, extract it from the stack.
+    # It is important to generate unique prefixes for different tests in the same class,
+    # so that the database namespace do not collide.
+    # 0th element on the stack is the current function. 1st element is the calling function.
+    if not test_name:
+        test_name = inspect.stack()[1][3]
+    prefix += test_name + '--'
```

Comment:
Line number 81 generates a 32-bit random number, that takes care of the same user running the tests multithreaded. A current time (second granularity) is also prefixed. So I doubt there will ever be a collision.

This part of generating prefixes need a cleanup. I will like to improve it a bit, so that all the tests could go to the S3 bucket in a directory like hierarchy. And artefacts from same tests could be bunched together. I also need to figure out a way to archive artefacts from failures. I have these items noted down, will file a ticket shortly.